Digitall.Warehouse.Application/Categories/Queries/GetCategoryByName/GetCategoryByNameQueryHandler.cs

```diff
@@ -0,0 +1,32 @@
+﻿using Digitall.Warehouse.Application.Abstractions.Messaging;
+using Digitall.Warehouse.Application.Abstractions.Persistence;
+using Digitall.Warehouse.Domain.Shared;
+
+namespace Digitall.Warehouse.Application.Categories.Queries.GetCategoryByName
+{
+    public class GetCategoryByNameQueryHandler(ICategoryRepository categoryRepository) 
+        : IQueryHandler<GetCategoryByNameQuery, GetCategoryByNameResponse>
+    {
+        private readonly ICategoryRepository _categoryRepository = categoryRepository;
+
+        public async Task<Result<GetCategoryByNameResponse>> Handle(
+            GetCategoryByNameQuery request,
+            CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrEmpty(request.Name))
+            {
+                return Result.Failure<GetCategoryByNameResponse>(Error.RequiredValue(nameof(request.Name)));
+            }
+
+            var category = await _categoryRepository.GetByNameAsync(request.Name);
+            if (category == null)
+            {
+                return Result.Failure<GetCategoryByNameResponse>(Error.NotFoundValue("Category", request.Name));
+            }
+
+            var getCategoryByNameResponse = new GetCategoryByNameResponse(category.Id, category.Name);
```

Comment:
I'll instroduce it in [PR4](https://github.com/tpilitis/Digitall.Warehouse/pull/4) or as additional PR. For category entity there is not need, because this is a record.
My DTOs will be represented as records and for such tiny one I cannot see the need from automapper.